cdap-common/src/test/java/io/cdap/cdap/io/SchemaHashTest.java

```diff
@@ -65,4 +65,13 @@ public void testSameSchemaHash() {
     Assert.assertEquals(schema.getSchemaHash(), schema.getSchemaHash());
     Assert.assertEquals(schemaWithLogicalType.getSchemaHash(), schemaWithLogicalType.getSchemaHash());
   }
+
+  @Test
+  public void testDecimalSchemaHash() {
+    Schema schema1 = Schema.decimalOf(3, 5);
+    Schema schema2 = Schema.decimalOf(3);
+
+    Assert.assertNotEquals(schema1.getSchemaHash(), schema2.getSchemaHash());
```

Comment:
It would be good to just do Assert.assertNotEquals(schema1, schema2) in addition to checking their hashes.

It looks like schema.equals compares using the schema hash. Do you happen to know why? It seems wrong unless the hash is somehow guaranteed to be unique?

Reply:
Added the check to the test. About schema hash I am not sure why we have it like that. The commit message is: `move a bunch of classes here`.

Reply:
Can you open a jira to investigate/follow up on the equals() question?

Reply:
sure, https://issues.cask.co/browse/CDAP-15355